AmplifyPlugins/Geo/Sources/AWSLocationGeoPlugin/Dependency/AWSDeviceTracker.swift

```diff
@@ -102,17 +103,16 @@ class AWSDeviceTracker: NSObject, CLLocationManagerDelegate, AWSDeviceTrackingBe
     }
     
     public func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
-        // if network is unreachable and `disregardLocationUpdatesWhenOffline` is set,
-        // don't store locations in local database
-        if options.disregardLocationUpdatesWhenOffline && !networkMonitor.networkConnected() {
-            return
-        }
         
         let currentTime = Date()
         // check if trackUntil time has elapsed
         if(options.trackUntil < currentTime) {
-            batchSendStoredLocationsToService(with: mapReceivedLocationsToPositions(receivedLocations: locations,
-                                                                                    currentTime: currentTime))
+            let receivedPositions = mapReceivedLocationsToPositions(receivedLocations: locations, currentTime: currentTime)
+            if let didUpdatePositions = options.locationProxyDelegate.didUpdatePositions {
+                batchSendStoredLocationsToProxyDelegate(with: receivedPositions, didUpdatePositions: didUpdatePositions)
+            } else {
+                batchSendStoredLocationsToService(with: receivedPositions)
```

Comment:
need to check for network state and `disregardLocationUpdatesWhenOffline` either send location updates or store locally here as well.